class.jetpack-autoupdate.php

```diff
@@ -42,11 +42,15 @@ private function __construct() {
 			add_action( 'shutdown', array( $this, 'log_results' ) );
 		}
 
-		// Anytime WordPress saves update data, we'll want to update our Jetpack option as well.
 		if ( is_main_site() ) {
+			// Anytime WordPress saves update data, we'll want to update our Jetpack option as well.
 			add_action( 'set_site_transient_update_plugins', array( $this, 'save_update_data' ) );
 			add_action( 'set_site_transient_update_themes', array( $this, 'save_update_data' ) );
 			add_action( 'set_site_transient_update_core', array( $this, 'save_update_data' ) );
+			// Anytime a connection to jetpack is made, sync the update data
+			add_action( 'update_option_jetpack_blog_token', array( $this, 'save_update_data' ) );
```

Comment:
Good idea :) but this won't work as `blog_token` is stored inside the option `jetpack_private_options`. I would add a new hook `do_action( 'jetpack_site_registered' )` to the `register` method inside `class.jetpack.php`.